src/analyze_logs.py

```diff
@@ -189,6 +189,10 @@ def log_file_p(file_name):
                 exit()
             time_range_info += [time_range]
         time_range_info = sorted(time_range_info, key=lambda k: k[0])
+
+    if args.out is not None and not os.path.isfile(args.out):
```

Comment:
What if the file doesn't exist yet -- does it fail? It shouldn't.

Reply:
thats a good point, I have forgotten that it creates the file if it does not exist. Will fix it.